libslide/src/bignum/sub.rs

```diff
@@ -0,0 +1,145 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::utils::vec_i8_into_u8;
+use crate::bignum::utils::vec_u8_into_i8;
+use crate::bignum::Bignum;
+use std::mem;
+use std::ops;
+
+impl ops::Sub for Bignum {
+    type Output = Bignum;
+
+    fn sub(self, other: Bignum) -> Bignum {
+        // make the decimal vectors the same size
+        // fill lhs and rhs vectors as well
+        let self_size: usize = self.dec.len();
+        let other_size: usize = other.dec.len();
+        let (res_int, lhs_int, res_dec, lhs_dec) = if self_size > other_size {
+            let mut lhs_dec = vec![0; self_size];
+            lhs_dec[..other_size].clone_from_slice(&other.dec[..other_size]);
+            (self.int, other.int, self.dec, lhs_dec)
+        } else {
+            let mut lhs_dec = vec![0; other_size];
+            lhs_dec[..self_size].clone_from_slice(&self.dec[..self_size]);
+            (other.int, self.int, other.dec, lhs_dec)
+        };
+
+        // now we need to make the bigger number the res dec
+
+        let mut lhs = Bignum {
+            is_neg: false,
+            dec: lhs_dec,
+            int: lhs_int,
+        };
+        let mut rhs = Bignum {
+            is_neg: false,
+            dec: res_dec,
+            int: res_int,
+        };
+        if lhs > rhs {
+            mem::swap(&mut lhs, &mut rhs);
+        }
+        // 1. Decimal
+
+        let mut res: i8;
+        let mut carry: i8 = 0;
+        let mut res_dec_asi8 = vec_u8_into_i8(rhs.dec);
+        let lhs_dec_asi8 = vec_u8_into_i8(lhs.dec);
+        let mut res_int_asi8 = vec_u8_into_i8(rhs.int);
+        let lhs_int_asi8 = vec_u8_into_i8(lhs.int);
+
+        for i in (0..res_dec_asi8.len()).rev() {
+            res = res_dec_asi8[i] - lhs_dec_asi8[i] - carry;
+            carry = (res < 0) as i8;
+            res += carry * 10;
+            res_dec_asi8[i] = res;
+        }
+
+        // 2. Integer
+        for i in 0..lhs_int_asi8.len() {
+            res = res_int_asi8[i] - lhs_int_asi8[i] - carry;
+            carry = (res < 0) as i8;
+            res += carry * 10;
+            res_int_asi8[i] = res;
+        }
+
+        // Propogate the rest of the carry
+        let mut i = lhs_int_asi8.len();
+        while i < res_int_asi8.len() {
+            res = res_int_asi8[i] as i8 - carry;
+            if res < 0 {
+                res += 10;
+                carry = 1;
+            } else {
+                carry = 0;
+            }
```

Comment:
Suggestion from before

```suggestion
carry = res < 0;
res += 10 * carry;
```